### ISSUE TITLE
Gitlab Provider: add support for filtering ci variables when querying them

### ISSUE THREAD

**Issue by rfelgent** [external-secrets/external-secrets#1460]
**Describe the solution you'd like**
A filter for "environment_scope" when retrieving ci variables from gitlab. 

**What is the added value?**
The key/name for a variable is re-used through different stages (dev, staging, prod and so on).  I would like use variable with a clear name (`username`) and avoid prefixing or suffixing them (`password_dev`, `password_staging`, `password_prod`).
**Give us examples of the outcome**

tbd.

**Observations (Constraints, Context, etc):**

The current implementation of the gitlab provider (gitlab.go) already has some hints about the filter functionality

```
...
func (g *Gitlab) GetSecret(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) ([]byte, error) {
	if utils.IsNil(g.client) {
		return nil, fmt.Errorf(errUninitalizedGitlabProvider)
	}
	// Need to replace hyphens with underscores to work with Gitlab API
	ref.Key = strings.ReplaceAll(ref.Key, "-", "_")
	// Retrieves a gitlab variable in the form
	// {
	// 	"key": "TEST_VARIABLE_1",
	// 	"variable_type": "env_var",
	// 	"value": "TEST_1",
	// 	"protected": false,
	// 	"masked": true
	data, _, err := g.client.GetVariable(g.projectID, ref.Key, nil) // Optional 'filter' parameter could be added later
	if err != nil {
		return nil, err
	}

...
```

In the past @knelasevero has pushed some things regarding gitlab provider. That's the reason, I would love to get feedback from you about this requirement. Maybe, you could give me some hints to push me in the right direction when preparing a MR?!

best regards

**Comment by rfelgent** [comment]
@domizei385  and @moolen thank you very much for providing the PR #1565 !

@moolen , when will this feature be released ?

**Comment by moolen** [comment]
Hey @rfelgent! I think we can do a patch release soon, probably in the next days.

I'm gonna close this one then as it's implemented with #1565.
